### PR TITLE
Library royalties and multi asyncCalls

### PR DESCRIPTION
**Pull request by sasurobert**
vmcommon prepared for multiple asyncCalls and for smart contract library royalties

### CODE REVIEW

File: codeMetadata.go [hunk 1]
```diff
@@ -5,6 +5,7 @@ const lengthOfCodeMetadata = 2
 const (
 	METADATA_UPGRADEABLE = 1
 	METADATA_PAYABLE     = 2
+	METADATA_READABLE    = 3
```
Comment (andrei-marinica) [inline codeMetadata.go]: Masking with 3 will also look at Upgradeable, with which it shares the byte. Should proably use 4 instead

File: codeMetadata.go [hunk 2]
```diff
@@ -23,6 +25,7 @@ func CodeMetadataFromBytes(bytes []byte) CodeMetadata {
 
 	return CodeMetadata{
 		Upgradeable: (bytes[0] & METADATA_UPGRADEABLE) != 0,
```
Comment (andrei-marinica) [inline codeMetadata.go:25]: Extract the byte indexes as constants too, e.g.
UPGRADEABLE_BYTE = 0
UPGRADEABLE_BIT_MASK = 1
etc.
I think it is easier to read

**Review by andrei-marinica — CHANGES_REQUESTED** [review]

File: codeMetadata.go (hunk 2 above)
Comment (andrei-marinica) [reply codeMetadata.go:25]: Also Go doesn't like ALL_CAPS, so maybe rename to
upgradeableByte = 0,
upgradeableBitMask = 1

Comment (sasurobert) [reply codeMetadata.go:25]: would leave as it is.

**Review by andrei-marinica — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]
maybe we should move the payable check on the 0-th byte

**Pull request merged**